--- nvidiabl-0.87/nvidiabl-module.c~	2015-04-19 13:27:17.493140224 +0200
+++ nvidiabl-0.87/nvidiabl-module.c	2015-04-19 13:26:41.479808163 +0200
@@ -214,7 +214,11 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39)
 	
 	for (iii = 0 ; iii < sizeof(backlight_type_ids) ; iii++) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
+		if (strncasecmp(bl_type, backlight_type_ids[iii].id, sizeof(bl_type)) == 0) {
+#else
 		if (strnicmp(bl_type, backlight_type_ids[iii].id, sizeof(bl_type)) == 0) {
+#endif
 			props.type = backlight_type_ids[iii].type;
 			printk(KERN_INFO "nvidiabl: backlight type is %s\n", backlight_type_ids[iii].id);
 		}
